pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -208,7 +208,7 @@ def __init__(
         self.monitor = monitor
         self.verbose = verbose
         self.save_last = save_last
-        self.save_top_k = save_top_k
+        self.save_top_k = int(save_top_k)
```

Comment:
is this to prevent existing users/configs to set it as None?

Reply:
That would be one advantage.

Also to make sure nobody is accidentally passing floats. This change is not really critical for this PR

Reply:
This is a breaking change right ? What if user don't want to save the best model but just the latest ?

Reply:
Might be better to raise an exception for float directly. What if a user provides -0.5 ?

Reply:
> This is a breaking change right ? What if user don't want to save the best model but just the latest ?

No, same behaviour as previously

> Might be better to raise an exception for float directly. What if a user provides -0.5 ?

We don't validate that all the inputs arguments types are the expected in across our codebase, typing linters should be the ones showing these potential errors.

The int conversion here is just meant so `1.0` becomes `1` and so on since we have checks for the exact values below. The change can be undone though

Reply:
```suggestion
        self.save_top_k = save_top_k
```